nemo/collections/asr/parts/submodules/ctc_greed_decoding.py

```diff
@@ -0,0 +1,202 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import List
+
+import torch
+
+from nemo.collections.asr.parts.utils import rnnt_utils
+from nemo.core.classes import Typing, typecheck
+from nemo.core.neural_types import HypothesisType, LengthsType, LogprobsType, NeuralType
+
+
+def pack_hypotheses(hypotheses: List[rnnt_utils.Hypothesis], logitlen: torch.Tensor,) -> List[rnnt_utils.Hypothesis]:
+
+    if logitlen is not None:
+        if hasattr(logitlen, 'cpu'):
+            logitlen_cpu = logitlen.to('cpu')
+        else:
+            logitlen_cpu = logitlen
+
+    for idx, hyp in enumerate(hypotheses):  # type: rnnt_utils.Hypothesis
+        hyp.y_sequence = torch.tensor(hyp.y_sequence, dtype=torch.long)
+
+        if logitlen is not None:
+            hyp.length = logitlen_cpu[idx]
+
+        if hyp.dec_state is not None:
+            hyp.dec_state = _states_to_device(hyp.dec_state)
+
+    return hypotheses
+
+
+def _states_to_device(dec_state, device='cpu'):
+    if torch.is_tensor(dec_state):
+        dec_state = dec_state.to(device)
+
+    elif isinstance(dec_state, (list, tuple)):
+        dec_state = tuple(_states_to_device(dec_i, device) for dec_i in dec_state)
+
+    return dec_state
+
+
+class GreedyCTCInfer(Typing):
+    """A greedy transducer decoder.
+
+    Provides a common abstraction for sample level and batch level greedy decoding.
+
+    Args:
+        decoder_model: rnnt_utils.AbstractRNNTDecoder implementation.
```

Comment:
Looks like this section docs need some update.

Reply:
Done, thanks for reminding. So many classes..